lib/sdn_epc/ofsh_calls.ex

```diff
@@ -2,14 +2,32 @@ defmodule SdnEpc.OfshCalls do
 
   require Logger
 
+  # :ofs_handler callbacks
+
   def init(_mode, _ip, datapatch_id, _features, _version, _connection, _opts) do
+    {:ok, ofpc_sup} = start_ofp_channel_sup()
+    open_ofp_channel ofpc_sup
     SdnEpc.Forwarder.subscribe_messages_from_switch datapatch_id, :packet_in
     {:ok, datapatch_id}
   end
 
   def handle_message(msg = {:packet_in, _xid, _body}, _datapatch_id)  do
-    Logger.info "Packet in message received" 
+    Logger.info "Packet in message received"
     IO.inspect msg
     :ok
   end
+
+  # Helper functions
+
+  defp start_ofp_channel_sup do
+    ofpc_sup_spec = Supervisor.Spec.supervisor(:ofp_channel_sup, [1])
```

Comment:
To it right, all the channel sups should be started under dedicated, `simple_one_for_one` supervisor. Now you have your forwarder and all the chanel_sups (per switch) under one supervisor.